src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/CounterPayload.cs

```diff
@@ -7,7 +7,12 @@
 namespace System.Diagnostics.Tracing
 {
     [EventData]
-    internal sealed class CounterPayload : IEnumerable<KeyValuePair<string, object?>>
+#if NATIVEAOT
+    public // On NativeAOT, this must be public to prevent it from getting reflection blocked.
```

Comment:
Unfortunately I think this is stripped specifically because it's in System.Private.CoreLib and even the normal attributes don't work. @MichalStrehovsky can you link the bug for this issue?

Reply:
#72570

Reply:
Thanks for linking the issue Jan.

We already have multiple ifdefs like this in S.D.Tracing since the .NET Native days, for example:

https://github.com/dotnet/runtime/blob/e029183d484dabf02c87140415a1632a101b4647/src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/TraceLogging/PropertyValue.cs#L16-L22

We'd clean them up if/when we get rid of reflection blocking.